src/Phinx/Console/Command/AbstractCommand.php

```diff
@@ -401,4 +401,73 @@ protected function getSeedTemplateFilename()
     {
         return __DIR__ . self::DEFAULT_SEED_TEMPLATE;
     }
+
+    /**
+     * Gets environment to use for command.
+     *
+     * @return string|null name of environment, null if could not be found
+     */
+    protected function getEnvironment(InputInterface $input, OutputInterface $output): ?string
+    {
+        $environment = $input->getOption('environment');
+
+        if ($environment === null) {
+            $environment = $this->getConfig()->getDefaultEnvironment();
+            $output->writeln('<comment>warning</comment> no environment specified, defaulting to: ' . $environment);
+        } else {
+            $output->writeln('<info>using environment</info> ' . $environment);
+        }
+
+        if (!$this->getConfig()->hasEnvironment($environment)) {
+            $output->writeln(sprintf('<error>The environment "%s" does not exist</error>', $environment));
+
+            return null;
+        }
+
+        return $environment;
+    }
+
+    /**
+     * Prints out general runtime information about loaded environment.
```

Comment:
This seems to be a validation routine. The bool return is ambiguous. There should be a validation routine that's separate from logging.